jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/DijkstraClosestFirstIterator.java

```diff
@@ -0,0 +1,247 @@
+     * vertex and will be limited to the connected component that includes that vertex.
+     * 
+     * @param graph the graph to be iterated.
+     * @param source the source vertex
+     */
+    public DijkstraClosestFirstIterator(Graph<V, E> graph, V source)
+    {
+        this(graph, source, Double.POSITIVE_INFINITY);
+    }
+
+    /**
+     * Creates a new radius-bounded iterator for the specified graph. Iteration will start at the
+     * specified start vertex and will be limited to the subset of the connected component which
+     * includes that vertex and is reachable via paths of weighted length less than or equal to the
+     * specified radius.
+     *
+     * @param graph the graph
+     * @param source the source vertex
+     * @param radius limit on weighted path length, or Double.POSITIVE_INFINITY for unbounded search
+     */
+    public DijkstraClosestFirstIterator(Graph<V, E> graph, V source, double radius)
+    {
+        this.graph = Objects.requireNonNull(graph, "Graph cannot be null");
+        this.source = Objects.requireNonNull(source, "Sourve vertex cannot be null");
+        if (radius < 0.0) {
+            throw new IllegalArgumentException("Radius must be non-negative");
+        }
+        this.radius = radius;
+        if (graph instanceof DirectedGraph) {
+            this.specifics = new DirectedSpecifics((DirectedGraph<V, E>) graph);
+        } else {
+            this.specifics = new UndirectedSpecifics(graph);
+        }
+        this.heap = new FibonacciHeap<>();
+        this.seen = new HashMap<>();
+
+        // initialize with source vertex
+        updateDistance(source, null, 0d);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean hasNext()
+    {
+        if (heap.isEmpty()) {
+            return false;
+        }
+        FibonacciHeapNode<QueueEntry> vNode = heap.min();
+        double vDistance = vNode.getKey();
+        if (radius < vDistance) {
+            heap.clear();
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public V next()
+    {
+        if (!hasNext()) {
+            throw new NoSuchElementException();
+        }
+
+        // settle next node
+        FibonacciHeapNode<QueueEntry> vNode = heap.removeMin();
+        V v = vNode.getData().v;
+        double vDistance = vNode.getKey();
+
+        // relax edges
+        for (E e : specifics.edgesOf(v)) {
+            V u = Graphs.getOppositeVertex(graph, e, v);
+            double eWeight = graph.getEdgeWeight(e);
+            if (eWeight < 0.0) {
+                throw new IllegalArgumentException("Negative edge weight not allowed");
+            }
+            updateDistance(u, e, vDistance + eWeight);
+        }
+
+        return v;
+    }
+
+    /**
+     * Return the paths computed by this iterator. Only the paths to vertices which are already
```

Comment:
We should mention somewhere that using Dijkstra as a SingleSourcePaths algorithm is more efficient than repeatedly invoking getPath(u,v) for the same u but different v. FloydWarshall for example doesn't have this advantage.